plugins/jobtype/src/azkaban/jobtype/HadoopSecureHiveWrapper.java

```diff
@@ -240,20 +240,19 @@ private static void populateHiveConf(HiveConf hiveConf, String[] args) {
     }
 
     Map<String, String> hiveVarMap = new HashMap<String, String>();
-    int index = 0;
-    for (; index < args.length; index++) {
+    for (int index = 0; index < args.length; index++) {
       if ("-hivevar".equals(args[index])) {
         String hiveVarParam = stripSingleDoubleQuote(args[++index]);
-
-        String[] tokens = hiveVarParam.split("=");
-        if (tokens.length == 2) {
-          String name = tokens[0];
-          String value = tokens[1];
-          logger.info("Setting hivevar: " + name + "=" + value);
-          hiveVarMap.put(name, value);
-        } else {
+        // Separate the parameter string at its first occurence of "="
+        int gap = hiveVarParam.indexOf("=");
+        if (gap == -1) {
           logger.warn("Invalid hivevar: " + hiveVarParam);
+          continue;
```

Comment:
It will just finish executing. This could itself should not make an exception thrown. However, missing parameters may cause an exception to be thrown later. Here, I'm simply mimicking previous behavior, except that I don't mark the queries in question as invalid.